typing/typeclass.ml

```diff
@@ -1631,7 +1634,17 @@ let class_description env sexpr =
   (expr, expr.cltyp_type)
 
 let class_declarations env cls =
-  type_classes true approx_declaration class_declaration env cls
+  let info, env =
+    type_classes true approx_declaration class_declaration env cls
+  in
+  let ids, exprs =
+    List.split
+      (List.map
+         (fun (id, _, _, _, _, _, _, _, _, _, _, {ci_expr}) -> id, ci_expr)
```

Comment:
That's a good idea, and I think it'd make a suitable project for a compiler hacking evening.  I've added it to the [wiki](https://github.com/ocamllabs/compiler-hacking/wiki/Things-to-work-on#bugsandfeatures).